### PR TITLE
Eahw 2539/update bbd tests

### PR DESCRIPTION
**Pull request by claire-guest**

### CODE REVIEW

File: src/test/resources/uk.gov.homeoffice.digital.sas.timecard.bdd/timecard.feature [hunk 1]
```diff
@@ -15,8 +15,121 @@ Feature: Timecard
...
+      """
+      {
+        "ownerId": "#{personaManager.getPersona('Trevor').id.toString}",
+        "timePeriodTypeId": "#{resourceHelper.getResourceId('Trevor', 'timecard','time-period-types','name=="Shift"')}",
+        "actualStartTime": "2022-11-16T10:00:00Z",
+        "actualEndTime": "2022-11-16T10:00:00Z"
+      }
+      """
+    When Trevor creates the invalid time-entries in the timecard service
+    Then the last response should have a status code of 200
```
Comment (stephenkaye-ho) [inline src/test/resources/uk.gov.homeoffice.digital.sas.timecard.bdd/timecard.feature:39]: Shouldn't this return a 400 and an error message

**Review by stephenkaye-ho — CHANGES_REQUESTED** [review]
Nice work. I have one issue that there's a test that I think is wrong because the existing behaviour could be wrong. Surely a timeperiod with a duration of zero is not valid.

File: src/test/resources/uk.gov.homeoffice.digital.sas.timecard.bdd/timecard.feature (hunk 1 above)
Comment (claire-guest) [reply src/test/resources/uk.gov.homeoffice.digital.sas.timecard.bdd/timecard.feature:39]: I realise I updated the wording of this, when with the current rules, this is valid. If this behaviour is incorrect, would fixing it come under this ticket, or a separate bugfix?

**Review by taomoh — DISMISSED** [review]

File: src/test/resources/uk.gov.homeoffice.digital.sas.timecard.bdd/timecard.feature (hunk 1 above)
Comment (claire-guest) [reply src/test/resources/uk.gov.homeoffice.digital.sas.timecard.bdd/timecard.feature]: Bug ticket raised here: https://collaboration.homeoffice.gov.uk/jira/browse/EAHW-2640

**Review by stephenkaye-ho — APPROVED** [review]

**Review by taomoh — APPROVED** [review]

**Pull request merged**